### PR TITLE
gnome-shell-extension-pop-shell: init at 1.0.0

### PR DESCRIPTION
**Pull request by Synthetica9**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/desktops/gnome-3/extensions/pop-shell/default.nix [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  makeFlags = [ "DESTDIR=$(out)" ];
+  buildInputs = [ glib nodePackages.typescript ];
+
+  postInstall = ''
+    mv $out/usr/* $out
+    rmdir $out/usr
+  '';
+
+  meta = with stdenv.lib; {
+    description = ''
```
Comment (SuperSandro2000) [inline pkgs/desktops/gnome-3/extensions/pop-shell/default.nix:23]: That is the long description. A short one is also needed.

Comment (Synthetica9) [reply pkgs/desktops/gnome-3/extensions/pop-shell/default.nix]: Added

File: pkgs/desktops/gnome-3/extensions/pop-shell/default.nix [hunk 2]
```diff
@@ -0,0 +1,33 @@
+{ stdenv, lib, fetchFromGitHub, glib, nodePackages }:
```
Comment (luc65r) [inline pkgs/desktops/gnome-3/extensions/pop-shell/default.nix:1]: You don't need `lib` here.

**Review by luc65r — CHANGES_REQUESTED** [review]

**Review by worldofpeace — CHANGES_REQUESTED** [review]
Commit message should have `init at` not @

**Comment by Synthetica9** [comment]
> Commit message should have `init at` not @

The commit message already had that, it was only the pr title that didn't. Changed it

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
Can you please squash the commits together?

File: pkgs/desktops/gnome-3/extensions/pop-shell/default.nix [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  '';
+
+  meta = with stdenv.lib; {
+    description = "A tiling extension for the GNOME Shell";
+    longDescription = ''
+      A keyboard-driven layer for GNOME Shell which allows for quick and
+      sensible navigation and management of windows. The core feature of Pop
+      Shell is the addition of advanced tiling window management.
+    '';
+    license = licenses.gpl3;
```
Comment (SuperSandro2000) [inline pkgs/desktops/gnome-3/extensions/pop-shell/default.nix:29]: homepage is missing here.

File: pkgs/desktops/gnome-3/extensions/pop-shell/default.nix [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+  makeFlags = [ "DESTDIR=$(out)" ];
+  buildInputs = [ glib nodePackages.typescript ];
+
+  postInstall = ''
+    mv $out/usr/* $out
+    rmdir $out/usr
+  '';
+
+  meta = with stdenv.lib; {
```
Comment (SuperSandro2000) [inline pkgs/desktops/gnome-3/extensions/pop-shell/default.nix:22]:
```suggestion
  meta = with lib; {
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
please fix the eval error.

**Comment by Synthetica9** [comment]
@grahamc Wait, there was already a PR. Anyways; these are the extensions they use. I updated them to the latest version but can't test that atm since I don't have a Gnome setup (same reason I removed myself as maintainer).

File: pkgs/desktops/gnome-3/extensions/pop-shell/default.nix [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    rev = version;
+    sha256 = "1ba1nrnk4cqgjx5mygqdkw74xlankrkiib9rw0vwkjcgv9bj024a";
+  };
+
+  makeFlags = [ "DESTDIR=$(out)" ];
+  buildInputs = [ glib nodePackages.typescript ];
+
+  postInstall = ''
+    mv $out/usr/* $out
+    rmdir $out/usr
```
Comment (benneti) [inline pkgs/desktops/gnome-3/extensions/pop-shell/default.nix:19]: if you add  ` makeFlags = [ "XDG_DATA_HOME=$(out)/share" ];` you don't need to move the files

**Review by benneti — COMMENTED** [review]
Isn't a patch like 
https://github.com/NixOS/nixpkgs/blob/b4f8e0120472735bd0d056f1e1c9c2a14408bdff/pkgs/desktops/gnome-3/extensions/pop-os-shell/fix-gjs.patch
necessary?

**Comment by genofire** [comment]
Solved in  #147542 - i believe we could close this PR now.
Maybe you add yourself to maintainers ;)

**Comment by Synthetica9** [comment]
> Solved in #147542 - i believe we could close this PR now. Maybe you add yourself to maintainers ;)

Great to hear, but I don't currently use Gnome. Thanks for picking this up though!

**Pull request closed without merging**